telekom-workflow-engine/src/main/java/ee/telekom/workflow/executor/WorkflowExecutorImpl.java

```diff
@@ -119,7 +119,7 @@ public void abortWorkflow( long woinRefNum ){
                 workflowInstanceService.updateHistory( woin.getRefNum(), woin.getHistory() );
                 workflowInstanceService.markAborted( woin.getRefNum() );
                 workflowInstanceService.unlock( woin.getRefNum() );
-                archiveService.archive( woin.getRefNum() );
+                archiveService.archive( woin.getRefNum(), -1 );
```

Comment:
If I remember correctly, we discussed to try to do
engineFactory.getGraph( woin.getWorkflowName(), woin.getWorkflowVersion() )
here too, and if the result is not null, then use the archive period from that, or if null then fall back to -1